examples/core_api/timm-post/post.md

```diff
@@ -0,0 +1,286 @@
+import determined as det
+
+...
+
+distributed = det.core.DistributedContext.from_torch_distributed()
+with det.core.init(distributed=distributed) as core_context:
+    main(core_context)
+
+```
+The `DistributedContext` we create here provides simple synchronization primitives used to coordinate Core API functionality across workers.  If you're running with one of our supported launchers, you can use the correspondng `from_` helper function to create one from the corresponding environment variables -- otherwise, you can [initialize one manually](https://github.com/determined-ai/determined/blob/master/harness/determined/_core/_distributed.py).
+
+Reporting a dictionary of metrics is now a single API call:
+```
+# The original script already takes care of aggregating metrics across workers.
+train_metrics = train_one_epoch(...)
+
+# Metrics can only be reported on rank 0, to avoid duplicate reports.
+if args.rank == 0:
+    core_context.train.report_training_metrics(latest_batch=latest_batch, metrics=train_metrics)
+```
+And similarly for validation metrics:
+```
+if args.rank == 0:
+	# Prefix metrics with val_ to distinguish from training metrics.
+    core_context.train.report_validation_metrics(
+        latest_batch=latest_batch,
+        metrics={"val_" + k: v for k, v in eval_metrics.items()},
+    )
+```
+
+
+If we rerun our experiment and check the overview page, we can now see our metrics nicely presented:
+
+<TODO: Metrics Image>
+
+
+# Step 3: Checkpointing
+The `timm` training script already has it's own built-in checkpointing system.  Rather than fully replace it, we'll instead see how we can gracefully layer Determined's checkpointing on top of it.  In a real use case, this would avoid breaking any integrations that rely on the existing functionality.
+
+First, we'll pull some metadata from `get_cluster_info()`:
+
+```
+info = det.get_cluster_info()
+# If running in local mode, cluster info will be None.
+if info is not None:
+    latest_checkpoint = info.latest_checkpoint
+    trial_id = info.trial.trial_id
+else:
+    latest_checkpoint = None
+    trial_id = -1
+```
+
+To save checkpoints, we'll hook into `timm`'s `CheckpointSaver` class with our own derived version:
+```
+class DeterminedCheckpointSaver(CheckpointSaver):
+    def __init__(self, trial_id, epoch_length, *args, **kwargs):
+        self.trial_id = trial_id
+        self.epoch_length = epoch_length
+        super().__init__(*args, **kwargs)
+
+    def _save(self, save_path, epoch, metric=None):
+        super()._save(save_path, epoch, metric)
+        checkpoint_metadata = {
+            "latest_batch": self.epoch_length * (epoch + 1),
+            "trial_id": self.trial_id,
+        }
+        with core_context.checkpoint.store_path(checkpoint_metadata) as (path, _):
+            shutil.copy2(save_path, path.joinpath("data"))
+```
+The essential Core API call here is `core_context.checkpoint.store_path`, which returns a context manager that:
```

Comment:
have you considered using core_context.checkpoint.upload() instead of store_path?  It's quite a bit more intuitive, and for medium-to-small checkpoints there's no real benefit to store_path.